managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1790,8 +1790,12 @@ void updateCursor(ManagedCursorImpl cursor, PositionImpl newPosition) {
         }
     }
 
-    PositionImpl startReadOperationOnLedger(PositionImpl position) {
-        long ledgerId = ledgers.ceilingKey(position.getLedgerId());
+    PositionImpl startReadOperationOnLedger(PositionImpl position, ReadEntriesCallback callback) {
+        Long ledgerId = ledgers.ceilingKey(position.getLedgerId());
+        if (null == ledgerId) {
+            callback.readEntriesFailed(new ManagedLedgerNotFoundException("read ledger failed, the ledgerID is null"), null);
```

Comment:
1) we should not use ManagedLedgerNotFoundExcception. ManagedLedgerNotFoundExcception is used when a managed ledger is not found, not a ledger in the managed ledger is not found

2) the error message should be improved. `The ceilingKey(K key) method is used to return the least key greater than or equal to the given key, or null if there is no such key.` that means in this managed ledger there is no ledger greater than or equal to `position.getLedgerId()`.